ssm/main.tf

```diff
@@ -1,64 +1,277 @@
-# -- Variables --
-# Remove this variable when modules support count
-# https://github.com/hashicorp/terraform/issues/953
-variable "enabled" {
-  default     = 1
-  description = "Whether this module is enabled (hack around modules not supporting count)"
+data "aws_caller_identity" "current" {
 }
 
-variable "region" {
-  default     = "us-west-2"
-  description = "AWS Region"
+# KMS keys
+data "aws_iam_policy_document" "kms_ssm" {
+  statement {
```

Comment:
Is this statement needed?   Am I reading this correctly that it gives the root user all permissions on any KMS key?

Reply:
This is a specific permission that is [required for any/all KMS keys](https://docs.aws.amazon.com/kms/latest/developerguide/key-policy-overview.html):
> When the principal in a key policy statement is an [AWS account principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_principal.html#principal-accounts) in the format `"arn:aws:iam::111122223333:root"`, the policy statement doesn't give permission to any IAM principal. Instead, it gives the AWS account permission to use IAM policies to delegate the permissions specified in the key policy. (A principal in `"arn:aws:iam::111122223333:root"` format does not represent the [AWS account root user](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_root-user.html), despite the use of "root" in the account identifier.)
We have this set up for every KMS key within this repo and others.